public_html/wp-content/mu-plugins/blocks/assets/src/sessions/edit.js

```diff
@@ -0,0 +1,142 @@
+/**
+ * External dependencies
+ */
+import { isUndefined, pickBy, split } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+const apiFetch = wp.apiFetch;
+const { withSelect } = wp.data;
+const { Component, Fragment } = wp.element;
+const { addQueryArgs } = wp.url;
+
+/**
+ * Internal dependencies
+ */
+import SessionsBlockControls from './block-controls';
+import SessionsInspectorControls from './inspector-controls';
+import GridToolbar from '../shared/grid-layout/toolbar';
+
+const blockData = window.WordCampBlocks.sessions || {};
+
+const SESSIONS_ICON = 'list-view';
+const MAX_POSTS = 100;
+
+const ALL_POSTS_QUERY = {
+	orderby  : 'title',
+	order    : 'asc',
+	per_page : MAX_POSTS,
+	_embed   : true,
+};
+
+const ALL_TERMS_QUERY = {
+	orderby  : 'name',
+	order    : 'asc',
+	per_page : MAX_POSTS,
+};
+
+class SessionsEdit extends Component {
+	constructor( props ) {
+		super();
+
+		this.state = {
+			allSessionPosts      : null,
+			allSessionTracks     : null,
+			allSessionCategories : null,
+		};
+	}
+
+	componentWillMount() {
+		this.isStillMounted = true;
+
+		const allSessionPosts = apiFetch( {
+			path: addQueryArgs( `/wp/v2/sessions`, ALL_POSTS_QUERY ),
+		} );
+		const allSessionTracks = apiFetch( {
+			path: addQueryArgs( `/wp/v2/session_track`, ALL_TERMS_QUERY ),
+		} );
+		const allSessionCategories = apiFetch( {
+			path: addQueryArgs( `/wp/v2/session_category`, ALL_TERMS_QUERY ),
+		} );
+
+		if ( this.isStillMounted ) {
```

Comment:
@coreymckrill , I'm probably missing something, but I don't see how this would ever be `false` at this point, so it seems like the condition is unnecessary.

Is it something to do with `apiFetch()` being asynchronous or a promise or something?

( note that there are a few other places where we do this, so if we change one we should change them all )

Reply:
I got the `this.isStillMounted` pattern from the Latest Posts block in Gute Core. I suspect the reasoning is that since it's working with promises, there's a chance that between the time that the fetch happens and when the promise is resolved, the block could have been deleted, and then it would be trying to manipulate variables that no longer exist.

Reply:
This approach may change when we address #63. Let's continue the conversation there.